Gems/ROS2/Code/Source/Camera/ROS2CameraSensorComponent.h

```diff
@@ -40,34 +41,36 @@ namespace ROS2
         static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
         static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
         ~ROS2CameraSensorComponent() override = default;
-        AZ_COMPONENT(ROS2CameraSensorComponent, "{3C6B8AE6-9721-4639-B8F9-D8D28FD7A071}", ROS2SensorComponent);
+
+        AZ_COMPONENT(ROS2CameraSensorComponent, "{3C6B8AE6-9721-4639-B8F9-D8D28FD7A071}", SensorBaseType);
         static void Reflect(AZ::ReflectContext* context);
 
         // AzToolsFramework::Components::EditorComponentBase overrides ..
         void Activate() override;
         void Deactivate() override;
 
         // CameraCalibrationRequestBus::Handler overrides ...
-        AZ::Matrix3x3 GetCameraMatrix() const override;
-        int GetWidth() const override;
-        int GetHeight() const override;
-        float GetVerticalFOV() const override;
+        [[nodiscard]] AZ::Matrix3x3 GetCameraMatrix() const override;
+        [[nodiscard]] int GetWidth() const override;
+        [[nodiscard]] int GetHeight() const override;
+        [[nodiscard]] float GetVerticalFOV() const override;
 
     private:
         //! Helper that adds an image source.
         //! @tparam CameraType type of camera sensor (eg 'CameraColorSensor')
         template<typename CameraType>
-        void AddImageSource()
+        void SetImageSource()
         {
-            auto cameraName = GetCameraNameFromFrame(GetEntity());
+            const auto cameraName = GetCameraNameFromFrame(GetEntity());
```

Comment:
```suggestion
            const auto& cameraName = GetCameraNameFromFrame(GetEntity());
```

Reply:
This will not work here, because GetCameraNameFromFrame returns by value. I keep it as it is.